addons/mail/static/src/new/core/persona_model.js

```
@@ -0,0 +1,61 @@
+/* @odoo-module */
+
+import { createLocalId } from "./thread_model.create_local_id";
+
+/**
+ * @typedef Data
+ * @property {import("@mail/new/core/guest_model").Guest} guest
+ * @property {import("@mail/new/core/partner_model").Partner} partner
+ */
+
+export class Persona {
+    partner;
+    guest;
+
+    /**
+     * @param {import("@mail/new/core/messaging").Messaging['store']} store
+     * @param {import("@mail/new/core/persona_model").Data} data
+     * @returns {import("@mail/new/core/persona_model").Persona}
+     */
+    static insert(store, data) {
+        const localId = data.partner
+            ? createLocalId("Partner", data.partner.id)
+            : createLocalId("Guest", data.guest.id);
+        let persona = store.personas[localId];
+        if (!persona) {
+            persona = new Persona();
+            persona._store = store;
+            store.personas[localId] = persona;
+            // Get reactive version.
+            persona = store.personas[localId];
+        }
+        Object.assign(persona, {
+            guest: data?.guest,
+            partner: data?.partner,
```

We should throw an error if both partner and guest are present?